frontend/packages/dev-console/src/components/project-access/ProjectAccessForm.tsx

```diff
@@ -0,0 +1,198 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik } from 'formik';
+import { k8sCreate, k8sKill, k8sPatch } from '@console/internal/module/k8s';
+import { RoleBindingModel } from '@console/internal/models';
+import { LoadingBox, PageHeading } from '@console/internal/components/utils';
+import { generateSecret } from '../import/import-submit-utils';
+import { withStartGuide } from '../../../../../public/components/start-guide';
+import {
+  validationSchema,
+  UserRole,
+  UserRoleBinding,
+  getRolesWithNameChange,
+  getNewRoles,
+  getRemovedRoles,
+  getfinalRoles,
+} from './projectAccessForm-utils';
+
+export interface ProjectAccessFormProps {
+  ProjectAccessFormComponent: React.ComponentType<any>;
+  formName: string;
+  namespace: string;
+  roleBindings?: { data: []; loaded: boolean };
+  projects?: { data: []; loaded: boolean };
+}
+
+const ProjectAccessForm: React.FC<ProjectAccessFormProps> = ({
+  ProjectAccessFormComponent,
+  formName,
+  namespace,
+  roleBindings,
+  projects,
+}) => {
+  if (!projects.loaded || (projects.loaded && !_.isEmpty(projects.data) && !roleBindings.loaded)) {
+    return <LoadingBox />;
+  }
+
+  const noProjectsAvailable = projects.loaded && _.isEmpty(projects.data);
+
+  const filteredRoleBindings = _.filter(
+    roleBindings.data,
+    (user: UserRole) => user.roleRef.name === 'view' || user.roleRef.name === 'admin',
```

Comment:
This should be maintained in a white list of roles we support so it's easier to re-use.

Reply:
Created enum for the roles in the utils file under `project-access`